web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts

```diff
@@ -29,15 +29,33 @@ const updateRowsIntegral = (oldRows: Map<number, Row>, deltaRows: Row[]) => {
   return oldRows
 }
 
-const computeAndSet = (prev: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
-  updateRowsIntegral(prev, parsedRows)
-  setRows(Array.from(prev.values()).sort((a, b) => a.genId - b.genId))
-  return prev
+const computeAndSet = (integral: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
+  setRows(old => {
+    const relationChanged = !old.length
```

Comment:
is this sufficient to indicate relation has changed? no false positives?

Reply:
The only false positive would be when on update relation didn't change but it was empty, in which case it's not a problem as the behavior of update is idempotent